src/module/wallet/utils/WalletController.ts

```diff
@@ -0,0 +1,245 @@
+import { NetworkType } from "module/settings/state/SettingsState";
+import { Chains } from "near-peersyst-sdk";
+import ServiceInstance from "../state/ServiceInstance/ServiceInstance";
+import { CreateInstanceReturn } from "../state/ServiceInstance/ServiceInstance.types";
+import { Wallet } from "../state/WalletState";
+import { SecureWalletInfo, UnencryptedWalletInfo } from "../wallet.types";
+import { WalletStorage } from "../WalletStorage";
+import { WalletUtils } from "./WalletUtils";
+
+interface WalletsControllerBaseReturn {
+    wallets: Wallet[];
+}
+
+interface TempWalletGroup {
+    deletedIds: number[];
+    privateKey: string;
+    newWallets: CreateInstanceReturn[];
+}
+
+export default class WalletController {
+    /**
+     * Import a wallet from a private key or a mnemonic
+     * @returns Returns the new wallets and if the secret is repeated
+     */
+    static async importWallets(
+        network: NetworkType,
+        pin?: string,
+        mnemonic?: string,
+        privateKeyParam?: string,
+    ): Promise<WalletsControllerBaseReturn> {
+        const storageWallets = await WalletStorage.getUnencryptedWallets(network);
+        const newWallets: Wallet[] = [];
+        const secureStorage = await WalletStorage.getSecure();
+        const walletIds: SecureWalletInfo["walletIds"] = []; //Wallets to be added to the secure storage
+
+        let privateKey = "";
+
+        if ((mnemonic && secureStorage?.mnemonic === mnemonic) || secureStorage?.[network].find((w) => w.privateKey === privateKeyParam)) {
+            return { wallets: [] };
+        }
+
+        const imported = !mnemonic || mnemonic !== secureStorage?.mnemonic;
+
+        //Init serviceInstanceMap
+        const accounts = await ServiceInstance.addServiceInstances({ network, privateKey: privateKeyParam, mnemonic });
+        const numOfPrevWallets = storageWallets.length;
+
+        //Add new accounts
+        for (const [index, { account, privateKey: pK }] of accounts.entries()) {
+            /**
+             * Don't used privateKeyParam because it can be undefined
```

Comment:
privateKeyParam is not used because it can be undefined*

Reply:
oki